### PR TITLE
Creation of secretPage function and funfact html page

### PR DESCRIPTION
**Pull request by J-coder13**

### CODE REVIEW

File: portfolio/src/main/webapp/gallery.html
Comment (saucepwn) [inline portfolio/src/main/webapp/gallery.html:12]: remove extra space

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js:35]: Multiline comments should be formatted like:
```
/*
 * This is a 
 * multi line comment
 */
```

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js:39]: nit: Google style would have the curly brace separated from the parameter list by a space:
```suggestion
function secretPage() {
```

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js]: This line is duplicated twice and should be moved to its own method to avoid code duplication

Comment (saucepwn) [inline portfolio/src/main/webapp/script.js]: this.a will create an instance variable attached to the page. It looks like this variable is only used in this function, so you can declare it as a local variable:

var a = ...
var b = ...

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js]: nit: Keep spacing consistent

```suggestion
    let hold = this.b;
```

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js]: Avoid double blank lines

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js]: Indent comments as you would code, so this should be indented +2 spaces.
```suggestion
  // choice determines which equation the user gets 
```

File: portfolio/src/main/webapp/style.css
Comment (saucepwn) [inline portfolio/src/main/webapp/style.css]: Keep indentation consistent throughout the file (2 spaces)

File: portfolio/src/main/webapp/talent.html
Comment (saucepwn) [inline portfolio/src/main/webapp/talent.html]: If this is a placeholder for the talent page, can you change the page title to identify it as such?

File: portfolio/src/main/webapp/talent.html
Comment (saucepwn) [inline portfolio/src/main/webapp/talent.html]: Remove errant space after the opening `<h1>` tag unless there's a need for it

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js]: This is an interesting way of doing this. It's good to demonstrate how to save functions to a variable (called closures in Javascript) but in this context it makes the code more complex and harder to read. Instead of saving these operations to function closures, you could directly do perform the operation in the switch statement on line 72

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js]: Here's a fun shorthand to collapse these three lines into one:

isCorrect = input == answer;

**Review by saucepwn — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/gallery.html
Comment (J-coder13) [reply portfolio/src/main/webapp/gallery.html:12]: Got it.

File: portfolio/src/main/webapp/script.js
Comment (J-coder13) [reply portfolio/src/main/webapp/script.js]: Got it.

Comment (J-coder13) [reply portfolio/src/main/webapp/script.js]: Got it

File: portfolio/src/main/webapp/script.js
Comment (J-coder13) [reply portfolio/src/main/webapp/script.js]: Got it

File: portfolio/src/main/webapp/style.css
Comment (J-coder13) [reply portfolio/src/main/webapp/style.css]: Got it

File: portfolio/src/main/webapp/talent.html
Comment (J-coder13) [reply portfolio/src/main/webapp/talent.html]: Got it.

File: portfolio/src/main/webapp/script.js
Comment (J-coder13) [reply portfolio/src/main/webapp/script.js]: Cool I definitely implement that.

File: portfolio/src/main/webapp/script.js
Comment (J-coder13) [reply portfolio/src/main/webapp/script.js]: Got it.

File: portfolio/src/main/webapp/talent.html
Comment (J-coder13) [reply portfolio/src/main/webapp/talent.html]: Got it.

File: portfolio/src/main/webapp/script.js
Comment (J-coder13) [reply portfolio/src/main/webapp/script.js]: Got it

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js]: There are two places where random numbers are required (one between [1...4], the other [1...200]). Can you make this function take an input parameter to specify the highest number so you can use it in both cases?

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js]: Instead of using `var`, which preceedes `const` and `let`, use either `const` or `let`. Since these values will be reassigned, `let` would be the right choice here.

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js]: Try to keep styling consistent
```suggestion
  if(b > a) {
```

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js:39]: Consistent styling
```suggestion
function secretPage() {
```

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js]: Spacing consistency
```suggestion
        input = prompt(a + ' + ' + b + '?', 0);
```

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js]: Spacing consistency
```suggestion
        input = prompt(a + ' - ' + b + '?', 0);
```

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js]: Spacing consistency
```suggestion
        input = prompt(a + ' * ' + b + '?', 0);
```

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js]: Spacing consistency
```suggestion
        input = prompt(a + ' / ' + b + '? Rounded to the nearest Integer', 0);
```

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js]: `secret` can be marked `const` instead of `let` since the value of `secret` is never reassigned

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js]: Spacing consistency
```suggestion
    if (isCorrect) {
```

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js]: Spacing consistency
```suggestion
    } else {
```

**Review by saucepwn — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/script.js
Comment (J-coder13) [reply portfolio/src/main/webapp/script.js]: Yeah that makes more sense.

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js:47]: `let` can be `const` here since the variable `hold` is not reassigned after initialization.

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js:56]: `choice` can be `const` since it is never reassigned after initialization.

File: portfolio/src/main/webapp/script.js
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js:42]: Per the Google Javascript style guide: https://google.github.io/styleguide/jsguide.html#formatting-horizontal-whitespace

> Horizontal Whitespace is required:
> After a comma (,) or semicolon (;). Note that spaces are never allowed before these characters.

```suggestion
  let a = randomNumber(200, 1);
  let b = randomNumber(200, 1);
```

**Review by saucepwn — APPROVED** [review]
Looking good, feel free to merge this PR right after addressing comments - no need for another review unless you have any unanswered questions.

**Pull request merged**
